wpt_common.py

```diff
@@ -141,3 +141,107 @@ def fetch_all_prs():
 
 def is_export_pr(pr):
     return bool(pr['chromium_commit'])
+
+
+def pr_number_from_tag(tagish):
+    """Extracts the PR number as integer from a string that begins with
+    merge_pr*, like merge_pr_6581 or merge_pr_6589-1-gc9db8d86f6."""
+    match = re.search(r'merge_pr_(\d+)', tagish)
+    if match is not None:
+        return int(match.group(1))
+    return None
+
+
+def get_tagged_prs():
+    """Gets the set of integers for which merge_pr_* exists."""
+    # --format="%(refname:strip=2) %(objectname)" would also include SHA-1
+    output = wpt_git(['tag', '--list', 'merge_pr_*'])
+    return set(pr_number_from_tag(tag) for tag in output.split() if tag)
+
+
+def git_contained_pr(commit):
+    """Returns the number of the latest contained PR of commit using git describe."""
+    try:
+        tag = wpt_git(['describe', '--tags', '--match', 'merge_pr_*', commit])
+        return pr_number_from_tag(tag)
+    except subprocess.CalledProcessError:
+        return None
+
+
+def pr_number(pr):
+    return int(pr['PR'])
+
+
+def pr_date(pr):
+    return dateutil.parser.parse(pr['merged_at'])
+
+
+def get_pr_latencies(prs, events=None, event_sha=None, event_date=None):
+    """For each PR, find the earliest event that included that PR,
+    and calucate the latencies between the PR and the event.
+
+    Args:
+        prs: list of PRs from fetch_all_prs()
+        events: list of events in any format
+        event_sha: function to get a sha from an event
+        event_date: function to get a datetime.datetime from an event
+
+    Returns list of { 'pr': pr, 'event': event, 'latency': latency } objects.
+    """
+
+    # Sort the PRs by merge date and filter out the ones that weren't tagged,
+    # since we don't know at which commit they were merged, and their computed
+    # latency can therefore be higher than the real latency. (Any PR which has
+    # correct merge information via the GitHub API should also have a tag:
+    # https://github.com/foolip/ecosystem-infra-stats/issues/6#issuecomment-375731858)
+    tagged_prs = get_tagged_prs()
+    prs = sorted(filter(lambda pr: pr_number(pr) in tagged_prs, prs),
+                 key=pr_date)
+
+    # We get PR-to-event latencies by the following process:
+    #  1. For each event find the latest contained PR. This is done using git
+    #     describe, and after this point the git commit graph doesn't matter.
+    #     Not all events necessarily have a known contained PR at all.
```

Comment:
When would it happen? Do we have an `event` so early that doesn't have any tagged commits before it?

Reply:
Yep, I ran into that when writing the code, possibly that was what caused me to backfill PRs, I'm not sure.